service/src/main/java/tennisclub/service/UserServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package tennisclub.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import tennisclub.dao.UserDao;
+import tennisclub.entity.User;
+import tennisclub.entity.enums.Role;
+
+import java.util.List;
+
+public class UserServiceImpl implements UserService {
+
+    private final UserDao userDao;
+
+    private  final PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+
+    @Autowired
+    public UserServiceImpl(UserDao userDao) {
+        this.userDao = userDao;
+    }
+
+
+    @Override
+    public void register(User user, String plainTextPassword) {
```

Comment:
When registering, shouldn't we also assign a default `User` role to this newly created instance? 🤔